components/dot_menu/dot_menu.jsx

```diff
@@ -295,78 +303,75 @@ export default class DotMenu extends Component {
+                    <Menu.ItemAction
+                        id={`unread_post_${this.props.post.id}`}
+                        show={!isSystemMessage}
+                        text={Utils.localizeMessage('post_info.unread', 'Mark as Unread')}
+                        onClick={this.handleUnreadMenuItemActivated}
+                    />
+                    <Menu.ItemAction
+                        id={`permalink_${this.props.post.id}`}
+                        show={!isSystemMessage}
+                        text={Utils.localizeMessage('post_info.permalink', 'Permalink')}
+                        onClick={this.handlePermalinkMenuItemActivated}
+                    />
+                    <Menu.ItemAction
+                        show={isMobile && !isSystemMessage && this.props.isFlagged}
+                        text={Utils.localizeMessage('rhs_root.mobile.unflag', 'Unflag')}
+                        onClick={this.handleFlagMenuItemActivated}
+                    />
+                    <Menu.ItemAction
+                        show={isMobile && !isSystemMessage && !this.props.isFlagged}
+                        text={Utils.localizeMessage('rhs_root.mobile.flag', 'Flag')}
+                        onClick={this.handleFlagMenuItemActivated}
+                    />
+                    <Menu.ItemAction
+                        id={`unpin_post_${this.props.post.id}`}
+                        show={!isSystemMessage && !this.props.isReadOnly && this.props.post.is_pinned}
+                        text={Utils.localizeMessage('post_info.unpin', 'Unpin')}
+                        onClick={this.handlePinMenuItemActivated}
+                    />
+                    <Menu.ItemAction
+                        id={`pin_post_${this.props.post.id}`}
+                        show={!isSystemMessage && !this.props.isReadOnly && !this.props.post.is_pinned}
+                        text={Utils.localizeMessage('post_info.pin', 'Pin')}
+                        onClick={this.handlePinMenuItemActivated}
+                    />
+                    {!isSystemMessage && (this.state.canEdit || this.state.canDelete) && this.renderDivider('edit')}
+                    <Menu.ItemAction
+                        id={`edit_post_${this.props.post.id}`}
+                        show={this.state.canEdit}
+                        text={Utils.localizeMessage('post_info.edit', 'Edit')}
+                        onClick={this.handleEditMenuItemActivated}
+                    />
+                    <Menu.ItemAction
+                        id={`delete_post_${this.props.post.id}`}
+                        show={this.state.canDelete}
+                        text={Utils.localizeMessage('post_info.del', 'Delete')}
+                        onClick={this.handleDeleteMenuItemActivated}
+                        isDangerous={true}
+                    />
+                    {(pluginItems.length > 0 || this.props.components[PLUGGABLE_COMPONENT]) && this.renderDivider('plugins')}
```

Comment:
isn't this removing the original separator for plugin actions? I'm not sure we should remove that one

Reply:
The original separator only existed if there were plugin actions, but it didn't exist if the user was using a `Pluggable` component (the previous iteration of plugin actions)